src/drivers/gitlab.js

```diff
@@ -271,31 +272,49 @@ class Gitlab {
       body
     });
 
-    if (autoMerge) {
-      await this.prAutoMerge({ mergeRequestId: iid });
-    }
-
+    if (autoMerge)
+      await this.prAutoMerge({ pullRequestId: iid, mergeMode: autoMerge });
     return url;
   }
 
   /**
-   * @param {{ mergeRequestId: string }} param0
+   * @param {{ pullRequestId: string }} param0
    * @returns {Promise<void>}
    */
-  async prAutoMerge({ mergeRequestId }) {
+  async prAutoMerge({ pullRequestId, mergeMode, mergeMessage = undefined }) {
+    if (mergeMode === 'rebase')
+      throw new Error(`Rebase auto-merge mode not implemented for GitLab`);
+
     const projectPath = await this.projectPath();
 
-    const endpoint = `/projects/${projectPath}/merge_requests/${mergeRequestId}/merge`;
+    const endpoint = `/projects/${projectPath}/merge_requests/${pullRequestId}/merge`;
     const body = new URLSearchParams();
-    body.append('merge_when_pipeline_succeeds', true);
+    body.set('merge_when_pipeline_succeeds', true);
+    body.set('squash', mergeMode === 'squash');
+    if (mergeMessage !== undefined)
```

Comment:
Do empty strings work well?
```suggestion
    if (mergeMessage)
```

Reply:
I don't know the API behavior for false values, so it's safe to do what you propose unless proven otherwise.